pyrate/core/orbital.py

```diff
@@ -81,82 +81,93 @@ def remove_orbital_error(ifgs: List, params: dict) -> None:
     files. The network method assumes the given ifgs have already been reduced
     to a minimum spanning tree network.
     """
-    mpiops.run_once(__orb_params_check, params)
     ifg_paths = [i.data_path for i in ifgs] if isinstance(ifgs[0], Ifg) else ifgs
+    degree = params[C.ORBITAL_FIT_DEGREE]
     method = params[C.ORBITAL_FIT_METHOD]
-    # mlooking is not necessary for independent correction in a computational sense
-    # can use multiple procesing if write_to_disc=True
+    orbfitlksx = params[C.ORBITAL_FIT_LOOKS_X]
+    orbfitlksy = params[C.ORBITAL_FIT_LOOKS_Y]
+    offset = params[C.ORBFIT_OFFSET]
+
+    # Sanity check of the orbital params
+    if type(orbfitlksx) != int or type(orbfitlksy) != int:
+        msg = f"Multi-look factors for orbital correction should be of type: int"
+        raise OrbitalError(msg)
+    if degree not in [PLANAR, QUADRATIC, PART_CUBIC]:
+        msg = "Invalid degree of %s for orbital correction" % C.ORB_DEGREE_NAMES.get(degree)
+        raise OrbitalError(msg)
+    if method not in [NETWORK_METHOD, INDEPENDENT_METHOD]:
+        msg = "Invalid method of %s for orbital correction" % C.ORB_METHOD_NAMES.get(method)
+        raise OrbitalError(msg)
+
+    # Give informative log messages based on selected options
+    log.info(f'Calculating {__degrees_as_string(degree)} orbital correction using '
+             f'{__methods_as_string(method)} method')
+    if orbfitlksx > 1 or orbfitlksy > 1:
+        log.info(f'Multi-looking interferograms for orbital correction with '
+                 f'factors of X = {orbfitlksx} and Y = {orbfitlksy}')
 
     if method == INDEPENDENT_METHOD:
-        log.info('Calculating orbital correction using independent method')
-        #TODO: implement multi-looking for independent orbit method
-        if params[C.ORBITAL_FIT_LOOKS_X] > 1 or params[C.ORBITAL_FIT_LOOKS_Y] > 1:
-            log.warning('Multi-looking is not applied in independent orbit method')
-        ifgs = [shared.Ifg(p) for p in ifg_paths] if isinstance(ifgs[0], str) else ifgs
+        ifg0 = shared.Ifg(ifg_paths[0]) if isinstance(ifg_paths[0], str) else ifg_paths[0]
+        original_dm = get_design_matrix(ifg0, degree, offset)
```

Comment:
I'm not sure about this. I think it is saying that the full-res DM from ifg0 (the first) will be used in the upscale of fullorb for every ifg? Could the ramifications of this be the cause of @tfuhrmann's observations about independent correction being different when using multi-looking factor > 1

Reply:
Disregard the above comment. The DM is the same for every ifg anyway.